sources/packages/api/src/route-controllers/models/pagination.dto.ts

```diff
@@ -0,0 +1,48 @@
+import { IsEnum, IsIn, IsOptional, Max, Min } from "class-validator";
+import { FieldSortOrder } from "../../utilities";
+
+/**
+ * Common parameters used when an API result
+ * must enable pagination and search options.
+ */
+abstract class PaginationOptionsAPIInDTO {
+  /**
+   * Field to be sorted.
+   */
+  abstract sortField?: string;
+  /**
+   * Order to be sorted.
+   */
+  @IsEnum(FieldSortOrder)
+  sortOrder: FieldSortOrder;
+  /**
+   * Page number.
+   */
+  @Min(0)
+  page: number;
+  /**
+   * Page size or records per page.
+   */
+  @Min(1)
+  @Max(50)
+  pageLimit: number;
+  /**
+   * Criteria to be used to filter the records.
+   */
+  @IsOptional()
+  searchCriteria?: string;
+}
+
+export class ApplicationPaginationOptionsAPIInDTO extends PaginationOptionsAPIInDTO {
+  @IsIn(["status", "applicationNumber"])
+  sortField?: string;
```

Comment:
what about IsOptional() ?

Reply:
Or else we can override it to be mandatory if it is mandatory for the implementing DTO.

![image](https://user-images.githubusercontent.com/54600590/170781665-b7c3fae1-41db-4b85-8eb9-4b840fec9f00.png)

Reply:
Yes, it would be good to have it added.

Reply:
I added it for now to keep it in sync with the service model also (`PaginationOptions`).